generate_hierarchical_cv_db/__main__.py

```diff
@@ -0,0 +1,81 @@
+from cv_library.compressor import Compressor
+from cv_library.loss_functions import sequence_similarity
+from cv_storage import ContextVectorDB
+import cv_hierarchical_storage as cvhs
+from pathlib import Path
+import torch
+import numpy as np
+from tqdm import tqdm
+import shutil
+
+DEVICE = "cuda:0" if torch.cuda.is_available() else "cpu"
+torch.set_default_device(DEVICE)
+torch.set_default_dtype(torch.float16)
+
+def get_all_headings(db: ContextVectorDB) -> list[tuple[str, str]]:
+    res = []
+    for article_title in db.get_article_titles():
+        for section_name in db.get_section_names(article_title):
+            res.append((article_title, section_name))
+    return res
+
+
+# Not used because need to figure out how to batch tensors of different sizes
+# def make_batches(xs, batch_size):
+#     for i in range(0, len(xs), batch_size):
+#         yield xs[i:i+batch_size]
+# 
+
+def main():
+    compressor_chkpt = "../rigel-data/hierarchical-compression-checkpoint-2024-10-30/attention_model.pt"
+    compressor = Compressor(compressor_chkpt)
+    #batch_size = 1024
+
+    input_db_path = Path("../rigel-data/context-vectors-2024-10-30/")
+    output_db_path = Path("../rigel-data/context-vectors-compressed-2024-10-30")
+
+    # Currently we don't support resuming
+    # To prevent inconsistent states, clean the DB before writing to it
+    try:
+        shutil.rmtree(output_db_path)
+    except FileNotFoundError:
+        pass
+    output_db_path.mkdir()
+
+
+    input_db = ContextVectorDB(input_db_path)
+    output_db_config = cvhs.DBConfig(
+            vec_max_size = 4096,
+            vec_min_size = 8,
```

Comment:
This value should be, by default, `cv_library.hierarchical_compression.Attention.NUM_HEADS`

Reply:
NUM_HEADS seem to be 32. I'm not sure how to use it here.

I've also changed the parameters to DBConfig. Now there's a single `level_sizes` parameter, which is specified in the `justfile` as:

```
--level_sizes 16,64,256,1024,4096
```

Does this work?